ios-app/UI/TestReportViewController.swift

```diff
@@ -251,13 +251,21 @@ class TestReportViewController: UIViewController {
             presentingViewController?.presentingViewController as? ContentDetailPageViewController {
             
             goToContentDetailPageViewController(contentDetailPageViewController)
+        } else if let customTestViewController =
+                    self.presentingViewController as? CustomTestGenerationViewController {
+            
+            goToCourseListView(customTestViewController)
         } else if exam == nil {
```

Comment:
This whole back button logic can be extracted.
It should follow this structure.
if (isPresentedFromView()) {    // Check better name from chatgpt
    goToSomeView()
}